### ISSUE TITLE
Async code is not well supported: test not wait for it to finished

### ISSUE THREAD

**Issue by kLiHz** [thecoderok/MeasureThat.net#87]
For example I have a `wait` funtion that returns a `Promise` which resolves after the timer runs out.

```javascript
function wait(ms) {
  return new Promise(res => setTimeout(() => { res(ms); }, ms));
}
```

If I tried using `await wait(100);` in test cases, it said that top-level await is not supported.

And it doesn't seem to wait until the promise is resolved when I just call `wait()` and don't `await` on it, since I'll get a similar benchmark between `wait(50)` and `wait(100)`. The benchmark is also very high, like several tens of thousands ops per second. If we consider an async code finished after its promise is returned, the benchmark should be less than 20 ops/sec for `wait(50)`, and less than 10 ops/sec for `wait(100)`.

It looks like [jsbench.me](https://jsbench.me/) supports measuring async code after it's finished if user checked the "defer" checkbox. It will mark the test done when `deferrered.resolve()` is called. We could chain a function to a promise and the test would wait wait on it.

```javascript
wait(50).then(() => { deferred.resolve(); })
```

And [jsbenchmark.com]() supports the use of `await`. I made [several tests about `wait` on it](https://jsbenchmark.com/#eyJjYXNlcyI6W3siaWQiOiJzMXk4ZkJvdW45REZnSm9NeFRTUHoiLCJjb2RlIjoiY29uc3Qgd2FpdCA9IChtcykgPT4gbmV3IFByb21pc2UocmVzID0-IHNldFRpbWVvdXQoKCkgPT4geyByZXMobXMpOyB9LCBtcykpO1xuXG5hd2FpdCB3YWl0KDUwKTsiLCJkZXBlbmRlbmNpZXMiOltdLCJuYW1lIjoiYXdhaXQgNTAgbXMifSx7ImlkIjoidXRxMHdnMF9oc256anFTa0JYbjVaIiwiY29kZSI6ImNvbnN0IHdhaXQgPSAobXMpID0-IG5ldyBQcm9taXNlKHJlcyA9PiBzZXRUaW1lb3V0KCgpID0-IHsgcmVzKG1zKTsgfSwgbXMpKTtcblxuYXdhaXQgd2FpdCgxMDApOyIsImRlcGVuZGVuY2llcyI6W10sIm5hbWUiOiJhd2FpdCAxMDAgbXMifSx7ImlkIjoiYmdZUC1ZSldjQkV3elRZV21PYUlrIiwiY29kZSI6ImNvbnN0IHdhaXQgPSAobXMpID0-IG5ldyBQcm9taXNlKHJlcyA9PiBzZXRUaW1lb3V0KCgpID0-IHsgcmVzKG1zKTsgfSwgbXMpKTtcblxuYXdhaXQgd2FpdCg1MDApOyIsImRlcGVuZGVuY2llcyI6W10sIm5hbWUiOiJhd2FpdCA1MDAgbXMifSx7ImlkIjoiNXFwZWU2WV83dDVOem5MUWNyWTM4IiwiY29kZSI6ImNvbnN0IHdhaXQgPSAobXMpID0-IG5ldyBQcm9taXNlKHJlcyA9PiBzZXRUaW1lb3V0KCgpID0-IHsgcmVzKG1zKTsgfSwgbXMpKTtcblxuYXdhaXQgd2FpdCgxMDAwKTsiLCJkZXBlbmRlbmNpZXMiOltdLCJuYW1lIjoiYXdhaXQgMTAwMCBtcyJ9XSwiY29uZmlnIjp7Im5hbWUiOiJUZXN0IGFzeW5jIC8gYXdhaXQiLCJwYXJhbGxlbCI6dHJ1ZSwiZ2xvYmFsVGVzdENvbmZpZyI6eyJkZXBlbmRlbmNpZXMiOltdfSwiZGF0YUNvZGUiOiIifX0);

**Comment by thecoderok** [comment]
this was implemented in https://github.com/thecoderok/MeasureThat.net/commit/f5a078d8b0909b1a2fe0f4e2af6c0933c1dda262
See sample test:

* https://measurethat.net/Benchmarks/Show/32502/2/async-test
* https://measurethat.net/Benchmarks/Show/32503/12/deferred-regular-async-hybrid-test
